editor/gui/create_dialog.cpp

```diff
@@ -620,7 +620,16 @@ String CreateDialog::get_selected_type() {
 	}
 
 	String type = selected->get_text(0).get_slicec(' ', 0);
-	return ClassDB::class_exists(type) ? type : String(selected->get_meta("_script_path", ""));
+	if (ClassDB::class_exists(type)) {
+		return type; // CPP type - from the core or GDExtensions
+	}
+
+	const EditorData::CustomType *custom_type = EditorNode::get_editor_data().get_custom_type_by_name(type);
+	if (custom_type != nullptr) {
+		return custom_type->name; // Types via EditorPlugin::add_custom_type()
```

Comment:
Actually you can replace it with
```suggestion
		return custom_type->script->get_path(); // Types via EditorPlugin::add_custom_type()
```
so #114367 is unnecessary.